src/python/pants/option/global_options.py

```diff
@@ -58,6 +58,11 @@ class OwnersNotFoundBehavior(Enum):
   error = "error"
 
 
+class UnmatchedAddressSpecBehavior(Enum):
+  ErrorOnAnyUnmatched = 'error-on-any-unmatched'
+  NoErrorOnUnmatchedGlobs = 'no-error-on-unmatched-globs'
```

Comment:
Nit: Calling this `NoErrorOnUnmatchedGlobs` is a bit confusing for me (because unmatched globs can also refer to the other globs, the `source` ones). Could we change the name to `NoErrorOnUnmatchedTargetSpecs` or something of the sort?